packages/legend-application-studio/src/stores/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -314,5 +320,34 @@ export class ServiceRegistrationState {
         'Service version can not be empty in Semi-interactive and Prod service type',
       );
     }
+
+    // validate service parameter multiplicities
+    const supportedServiceParamMultiplicties = [
+      TYPICAL_MULTIPLICITY_TYPE.ONE,
+      TYPICAL_MULTIPLICITY_TYPE.ZEROMANY,
+      TYPICAL_MULTIPLICITY_TYPE.ZEROONE,
+    ].map((p) =>
+      this.editorStore.graphManagerState.graph.getTypicalMultiplicity(p),
+    );
+    const invalidParams = buildLambdaVariableExpressions(
+      (this.service.execution as PureExecution).func,
+      this.editorStore.graphManagerState,
+    )
+      .filter(filterByType(VariableExpression))
+      .filter(
+        (p) => !supportedServiceParamMultiplicties.includes(p.multiplicity),
```

Comment:
don't think we can always assume multiplicity comes from the same graph and the multiplicity index. The most appropriate check here is to do the comparison by `lower` and `upper` bounds. The fact we cache multiplicities in the graph is just small optimization, we don't require other coders to have to know about it. I think it's best to create a comparator helper for multiplicity and use this. I almost guarantee this might cause a bug some day 🤷

Reply:
yeah in terms of bugs i think we could just use comparator. we should try to always use the getMultiplicity though so they are still the same object. 
but in this casse if they happen to be different objects we should still not allow the ones with unsupported multiplicities.